libs/novui/src/json-schema-components/widgets/InputEditorWidget.tsx

```diff
@@ -0,0 +1,264 @@
+import { FocusEvent, ChangeEvent, useEffect, useMemo, useRef, useState } from 'react';
+
+import { ErrorSchema, WidgetProps } from '@rjsf/utils';
+
+import { Input } from '@mantine/core';
+import { RichTextEditor } from '@mantine/tiptap';
+
+import {
+  type Extensions,
+  NodeViewContent,
+  NodeViewProps,
+  NodeViewWrapper,
+  ReactNodeViewRenderer,
+  useEditor,
+} from '@tiptap/react';
+import Text from '@tiptap/extension-text';
+import Paragraph from '@tiptap/extension-paragraph';
+import { ReactRenderer, mergeAttributes } from '@tiptap/react';
+import Document from '@tiptap/extension-document';
+import { Mention as ExternalMention } from '@tiptap/extension-mention';
+
+import { css, cx } from '../../../styled-system/css';
+import { input, inputEditorWidget } from '../../../styled-system/recipes';
+import { splitCssProps } from '../../../styled-system/jsx';
+
+import { VariableSuggestionList, SuggestionListRef, VariableItem } from './VariableSuggestionList';
+
+const inputEditorClassNames = inputEditorWidget();
+
+const AUTOCOMPLETE_OPEN_TAG = '{{';
+const AUTOCOMPLETE_CLOSE_TAG = '}}';
+
+const VALID_VARIABLE_REGEX = new RegExp(`${AUTOCOMPLETE_OPEN_TAG}([^{}]*?)${AUTOCOMPLETE_CLOSE_TAG}`, 'g');
+
+// To get initial invalid syntax of {{..} or {{..<space>
+const INVALID_VARIABLE_REGEX = new RegExp(
+  `(${AUTOCOMPLETE_OPEN_TAG}[^{}|\\s]*[}|\\s](?!})|{{[^{}]*}(?!}})|{{[^{}]*$)`,
+  'g'
+);
+const AUTOCOMPLETE_REGEX_V2 = new RegExp(`${AUTOCOMPLETE_OPEN_TAG}(.*?(.*?))${AUTOCOMPLETE_CLOSE_TAG}`, 'm');
+
+const getInitContentWithVariableNodeView = (inputString: string) => {
+  let result = inputString?.toString().replace(VALID_VARIABLE_REGEX, (_, validContent) => {
+    return `<mention-component data-label="${validContent}" data-id="${validContent}"></mention-component>`;
+  });
+
+  result = result?.replace(INVALID_VARIABLE_REGEX, (match) => {
+    return `<mention-component data-label="${match}" data-id="${match}" data-error="true"></mention-component>`;
+  });
+
+  return result;
+};
+
+export function Mention({ editor, node, updateAttributes, ...props }: NodeViewProps) {
+  const [hasError, setHasError] = useState(node.attrs.error || false);
+
+  const updateVariableAttributes = (label: string, error: boolean) => {
+    updateAttributes({ label, error: `${error}` });
+    setHasError(error);
+  };
+
+  //In case of an invalid variable input, set content to be the label variable  which will include the whole invalid variable
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
☑ **chore:** Missing space‏
```suggestion
  // In case of an invalid variable input, set content to be the label variable  which will include the whole invalid variable
```